components/AddPopover/AddPopover.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { useState, useEffect, ReactElement } from "react";
+import styles from "./AddPopover.module.css";
+import { Button, Popover, Box, TextField, Autocomplete, Radio, RadioGroup, FormControlLabel} from "@mui/material";
+import Icon from "../../assets/Icon";
+import { Transaction, User } from "../../types/schema";
+import { addTransaction } from "../../firebase/firestore/transaction";
+
+type AddPopoverProps = {
+    allUsers: User[];
+    addAnchor: Element;
+    closeAdd: Function;
+    popoverid: string;
+};
+
+export const AddPopover: React.FunctionComponent<AddPopoverProps> = ({
+  allUsers,
+  addAnchor,
+  closeAdd,
+  popoverid,
+}: AddPopoverProps) => {
+    const [selectedUser, setSelectedUser] = useState('Select User')
+    const [addUser, setAddUser] = useState(null);
+
+    const [success, setSuccess] = useState(false);
+    const [addPoints, setAddPoints] = useState('10');
+    const [addType, setAddType] = useState('');
+    const [addMessage, setAddMessage] = useState('');
+
+    const handleAddConfirm = async () => {
+        //handle post request
+        const adding = {
+            admin_name: 'current Admin',
+            date: new Date(),
+            description: addMessage,
+            family_id: addUser.family_id,
+            point_gain: addType == 'redeem' ? -parseInt(addPoints) : parseInt(addPoints),
+            user_name: addUser.full_name,
+        }
+        addTransaction(adding as Transaction);
+
+        setSuccess(true);
+    }
+
+    const sleep = (ms) => {
```

Comment:
After thinking about why you need this `sleep` function, I think you could replace it with useEffect hooks! We want to update addUser based on the value of addAnchor, and useEffect is a great hook for that! We could put down addAnchor as a dependency and this will fire the callback every time addAnchor is changed, so then you can appropriately change the values of success as well!